File: website/src/main/java/app/api/ProductAJAXWebService.java
package app.api;

import app.api.product.CreateProductConfigResponse;
import app.api.product.GetProductResponse;
import app.api.product.ListProductResponse;
import core.framework.api.web.service.GET;
import core.framework.api.web.service.Path;
import core.framework.api.web.service.PathParam;

public interface ProductAJAXWebService {
    @GET
    @Path("/ajax/product")
    ListProductResponse list();

    @GET
    @Path("/ajax/product/create-config")
    CreateProductConfigResponse createConfig();

    @GET
    @Path("/ajax/product/:id")
    GetProductResponse get(@PathParam("id") String id);

    @GET
    @Path("/ajax/product/:id/child/:childId")
    GetProductResponse getChild(@PathParam("id") String id, @PathParam("childId") String childId);
}

File: website-frontend-ts/src/service/ProductAJAXWebService.ts
import {ListProductResponse, GetProductResponse, CreateProductConfigResponse} from "type/api";
import {ajax} from "core-fe";

export class ProductAJAXWebService {
    static list(): Promise<ListProductResponse> {
        return ajax("GET", "/ajax/product", {}, null);
    }
    static get(id: string): Promise<GetProductResponse> {
        return ajax("GET", "/ajax/product/:id", {id}, null);
    }
    static getChild(id: string, childId: string): Promise<GetProductResponse> {
        return ajax("GET", "/ajax/product/:id/child/:childId", {id, childId}, null);
    }
    static createConfig(): Promise<CreateProductConfigResponse> {
        return ajax("GET", "/ajax/product/create-config", {}, null);
    }
}
